bin/Debugger.Sample/Debugger.Sample.cpp

```diff
@@ -131,15 +160,32 @@ JsErrorCode RunScript(const wchar_t* filename, JsValueRef* result)
         return JsErrorInvalidArgument;
     }
 
-    // Load the script from the disk.
-    std::wstring script = LoadScript(fullPath);
-    if (script.empty())
+    if (loadScriptUsingBuffer)
     {
-        return JsErrorInvalidArgument;
+        JsValueRef sourceUrl;
+        JsValueRef scriptValueRef;
+
+        // Load script from the file
+        std::string script = LoadScriptUtf8(fullPath);
+        if (script.empty())
+        {
+            return JsErrorInvalidArgument;
+        }
+
+        IfFailRet(JsPointerToString(fullPath, wcslen(fullPath), &sourceUrl));
+        IfFailRet(JsCreateExternalArrayBuffer((void*) script.c_str(), (unsigned int) script.length(), NULL, NULL, &scriptValueRef));
```

Comment:
The lifetime here is suspect.  When you create an external array buffer it expects the backing data to be available via the buffer for the entire lifetime of the object.  In this case the backing data will be freed when `script` goes out of scope.  I think you'll need to create a class to own the data, create an instance on the heap, and provide a finalizer to delete the data when the array buffer is collected.

Reply:
Fixed